oxd-server/src/main/java/org/gluu/oxd/server/service/SchedulerService.java

```diff
@@ -0,0 +1,61 @@
+package org.gluu.oxd.server.service;
+
+import com.google.common.collect.Lists;
+import com.google.inject.Inject;
+import org.gluu.oxd.common.CoreUtils;
+import org.gluu.oxd.server.persistence.service.PersistenceService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+public class SchedulerService {
+
+    private static final Logger LOG = LoggerFactory.getLogger(SchedulerService.class);
+
+    private ConfigurationService configurationService;
+    private PersistenceService persistenceService;
+    private KeyGeneratorService keyGeneratorService;
+
+    @Inject
+    public SchedulerService(ConfigurationService configurationService, PersistenceService persistenceService, KeyGeneratorService keyGeneratorService) {
+        this.configurationService = configurationService;
+        this.persistenceService = persistenceService;
+        this.keyGeneratorService = keyGeneratorService;
+    }
+
+    public void scheduleTasks() {
+        ScheduledExecutorService scheduledExecutorService = CoreUtils.createExecutor();
+        scheduledExecutorService.scheduleWithFixedDelay(new Runnable() {
+            public void run() {
+                dbCleanUpTask();
+                jwksRegenerationTask();
+            }
+        }, configurationService.get().getDbCleanupIntervalInHours(), configurationService.get().getDbCleanupIntervalInHours(), TimeUnit.HOURS);
+    }
+
+    public void jwksRegenerationTask() {
+        if (configurationService.get().getEnableJwksGeneration()) {
```

Comment:
Minor but early return is always more readable due to inner `ifs`

```
if (!configurationService.get().getEnableJwksGeneration()) {
    return;
}
```

Reply:
Done